lib/msf/core/payload/windows/reverse_http.rb

```diff
@@ -44,31 +41,27 @@ def initialize(*args)
   #
   # Generate the first stage
   #
-  def generate
-    # Generate the simple version of this stager if we don't have enough space
-    if self.available_space.nil? || required_space > self.available_space
-      return generate_reverse_http(
-        ssl:  false,
-        host: datastore['LHOST'],
-        port: datastore['LPORT'],
-        url:  generate_small_uri,
-        retry_count: datastore['StagerRetryCount'])
-    end
-
+  def generate(opts={})
     conf = {
-      ssl:  false,
-      host: datastore['LHOST'],
-      port: datastore['LPORT'],
-      url:  generate_uri,
-      exitfunk: datastore['EXITFUNC'],
-      proxy_host: datastore['PayloadProxyHost'],
-      proxy_port: datastore['PayloadProxyPort'],
-      proxy_user: datastore['PayloadProxyUser'],
-      proxy_pass: datastore['PayloadProxyPass'],
-      proxy_type: datastore['PayloadProxyType'],
+      ssl:         opts[:ssl] || false,
+      host:        datastore['LHOST'],
+      port:        datastore['LPORT'],
+      url:         generate_small_uri,
       retry_count: datastore['StagerRetryCount']
     }
 
+    # Add extra options if we have enough space
+    unless self.available_space.nil? || required_space > self.available_space
```

Comment:
I find `unless` with more than a single predicate confusing. IMO, this would be more readable as:
```ruby
if self.available_space && required_space <= self.available_space
```